src/common/cli_wrapper.h

```diff
@@ -209,37 +275,47 @@ class CLIWrapper {
                           T val,
                           bool defaulted,
                           bool addToConfig) {
-    // std::cerr << "CLI::add(" << key << ") as vector" << std::endl;
-
+    // define YAML entry if requested
     if(addToConfig)
       config_[key] = val;
+    // create variable for the option
     vars_.insert(std::make_pair(key, std::make_shared<any_type>(val)));
 
+    // callback function collecting command-line arguments
     CLI::callback_t fun = [this, key](CLI::results_t res) {
-      // std::cerr << "CLI::callback(" << key << ") " << std::endl;
+      // get vector variable associated with the option
       auto &vec = vars_[key]->as<T>();
       vec.clear();
       bool ret = true;
+      // populate the vector with parser results
       for(const auto &a : res) {
         vec.emplace_back();
         ret &= CLI::detail::lexical_cast(a, vec.back());
       }
+      // update YAML entry
       config_[key] = vec;
       return (!vec.empty()) && ret;
     };
 
     auto opt = app_->add_option(args, fun, help);
-    opt->type_name(CLI::detail::type_name<T>())->type_size(-1);
+    // set human readable type value: VECTOR and
+    opt->type_name(CLI::detail::type_name<T>());
```

Comment:
Is this a setter? It reads like a getter. If this is our own code, I would prefer `setTypeName`, same for size.